nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -100,20 +107,25 @@ public static void checkDocumentEventContext(Event event, DocumentModel expected
      */
     public static void checkReceivedMail(List<MailMessage> mails, DocumentModel commentDocModel,
             DocumentModel commentedDocModel, Event event, String commentEventType) {
+        DocumentModel sourceDocument = ((DocumentEventContext) event.getContext()).getSourceDocument();
         int expectedMailsCount = COMMENT_REMOVED.equals(commentEventType) ? 0 : 1;
+        expectedMailsCount = COMMENT_ADDED.equals(commentEventType)
+                && !sourceDocument.getId().equals(commentedDocModel.getId()) //
+                        ? 2
+                        : expectedMailsCount;
         assertEquals(expectedMailsCount, mails.size());
 
         if (expectedMailsCount > 0) {
             String subject = String.format("[Nuxeo5]%s comment on '%s'",
-                    COMMENT_ADDED.equals(event.getName()) ? "New" : "Updated", commentedDocModel.getName());
+                    COMMENT_ADDED.equals(event.getName()) ? "New" : "Updated", sourceDocument.getName());
```

Comment:
We should keep the line as it was before because we want to assert against what we give to this method and not what is actually computed by service.